src/gc-stock.c

```diff
@@ -305,6 +305,9 @@ STATIC_INLINE void gc_setmark_pool(jl_ptls_t ptls, jl_taggedvalue_t *o,
 STATIC_INLINE void gc_setmark(jl_ptls_t ptls, jl_taggedvalue_t *o,
                               uint8_t mark_mode, size_t sz) JL_NOTSAFEPOINT
 {
+    if (o->bits.in_image) {
+        return;
+    }
```

Comment:
Why is this true? Do we never sweep objects in images?

Reply:
This is a drive by fix here. Images are swept by different code. And the force old code found a bug here

Reply:
Can we split that fix out?